### PR TITLE
Parent fix

### PR DESCRIPTION
**Pull request by kaszim**
## Description
This fixes a couple of issues that appeared because video frames were not parented. Firstly they will now properly be css:ed. Also this improved performance greatly.

While at it I tested some vlc flags, and found that opengl performed better than directx, so I changed that too.

## Important files to review
main.py
videoframegrid.py
videoframes.py

### CODE REVIEW

File: videoframes.py [hunk 1]
```diff
@@ -196,9 +196,9 @@ class RewindedVideoFrame(_VideoFrame):
             played from.
     """
 
-    def __init__(self, vlc_instance, stream_buffer, parent):
-        super(RewindedVideoFrame, self).__init__(vlc_instance)
-        # Set the parent
+    def __init__(self, parent, vlc_instance, stream_buffer):
+        super(RewindedVideoFrame, self).__init__(None, vlc_instance)
```
Comment (kaszim) [inline videoframes.py]: I do not add a parent, as we want a new window to be born.

**Comment by kaszim** [comment]
#93 Will have a easier time skinning now.

File: main.py [hunk 2]
```diff
@@ -35,8 +35,8 @@ def __init__(self):
         # Connect threading signals
         self.add_frame.connect(self.setup_videoframe)
         self.fail_add_stream.connect(self.on_fail_add_stream)
-
-        self.vlc_instance = vlc.Instance("--no-xlib")
+        # Opengl performs better on windows, which is odd
```
Comment (svenanderzen) [inline main.py]: Perhaps we shouldn't have this comment in the source? A more informative comment about the options would be preferred though:
```
self.vlc_instance = vlc.Instance("
    --no-xlib                     # Turn off XInitThreads()
    --vout=opengl            # Force OpenGL as video output module for better performance on Windows
    --avcodec-hw=any     # Hardware decoding
    --avcodec-threads=0 # Number of threads used for decoding, 0 meaning auto
")
```

File: videoframegrid.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def _add_videoframe(self, videoframe):
 
     def _create_videoframe(self, vlc_instance, stream_url, stream_options, quality):
         """Creates a new LiveVideoFrame object."""
-        return LiveVideoFrame(vlc_instance, stream_url, stream_options, quality)
+        return LiveVideoFrame(self.parent, vlc_instance, stream_url, stream_options, quality)
```
Comment (svenanderzen) [inline videoframegrid.py]: Perhaps it had something to do with the CSS?

Comment (kaszim) [inline videoframegrid.py]: As `VideoFrameGrid` in turn is a `QGridLayout`, which is _not_ a `QWidget`, it cannot be used as a parent for other `QWidget` objects.

File: main.py (hunk 2 above)
Comment (kaszim) [inline main.py]: Oh yes 🥇

File: main.py [hunk 4]
```diff
@@ -35,8 +35,8 @@ def __init__(self):
...
+        self.vlc_instance = vlc.Instance("--no-xlib --vout=opengl --avcodec-hw=any --avcodec-threads=0")
```
Comment (kaszim) [inline main.py]: Yep no performance gain after testing again. Which also makes sense as we are using callbacks (they block hardware decoding).

**Comment by kaszim** [comment]
@svenanderzen Could you try creating a `vlc_instance` in the container instead? For me I reduce cpu usage by 10 % units. Feels like I am seeing things that are not real though 😨

File: videoframegrid.py (hunk 3 above)
Comment (svenanderzen) [inline videoframegrid.py]: Oh, nice call! 👍

**Comment by kaszim** [comment]
@svenanderzen Can you review the changes? As alot of ui functioanlity depends on the fixes introduced here. Oh and why is travis failing my builds? I don't see vlc imported in main 😨 @mictab help pls

**Comment by mictab** [comment]
@kaszim It's imported here 😊 https://github.com/kaszim/desktop-stream-viewer/blob/4b82963a2d713964f1aceb28810d0991465efdf7/main.py#L237

**Comment by svenanderzen** [comment]
LGTM

**Pull request merged**